MacOS Clang Build Error [-Wunused-private-field] in numa_memory_resource.cpp

In file [numa_memory_resource.hpp](https://github.com/normanrz/zweirise/blob/ef627969a72b76eccea5674193d77058b620ab99/src/lib/utils/numa_memory_resource.hpp) the variable `_alignment` is unused, if flag `OPOSSUM_NUMA_SUPPORT` is not set.
This causes a build error on MacOS with clang.
This file was added with PGASUS NUMA MR #216 .

I would resolve this by only defining the variable, if the flag is set.

Error Message
```
In file included from /Users/fabian/git/zweirise/src/lib/utils/numa_memory_resource.cpp:1:
/Users/fabian/git/zweirise/src/lib/utils/numa_memory_resource.hpp:29:16: error: private field '_alignment' is not used [-Werror,-Wunused-private-field]
  const size_t _alignment = 1;
               ^
1 error generated.
make[2]: *** [src/lib/CMakeFiles/opossum.dir/utils/numa_memory_resource.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/lib/CMakeFiles/opossum.dir/all] Error 2
make: *** [all] Error 2
```

--- a/src/lib/utils/numa_memory_resource.hpp
+++ b/src/lib/utils/numa_memory_resource.hpp
@@ -25,8 +25,8 @@ class NUMAMemoryResource : public boost::container::pmr::memory_resource {
  private:
 #if OPOSSUM_NUMA_SUPPORT
   const numa::MemSource _mem_source;
-#endif
   const size_t _alignment = 1;
+#endif
 };
 
 }  // namespace opossum

